src/sagemaker/multidatamodel.py

```diff
@@ -0,0 +1,308 @@
+
+MULTI_MODEL_CONTAINER_MODE = "MultiModel"
+
+
+class MultiDataModel(Model):
+    """A SageMaker ``MultiDataModel`` that can be used to deploy multiple models to the same
+    SageMaker ``Endpoint``, and also deploy additional models to an existing SageMaker
+    multi-model ``Endpoint``
+    """
+
+    def __init__(
+        self,
+        name,
+        model_data_prefix,
+        model=None,
+        image=None,
+        role=None,
+        sagemaker_session=None,
+        **kwargs
+    ):
+        """Initialize a ``MultiDataModel``. In addition to these arguments, it supports all
+           arguments supported by ``Model`` constructor
+
+        Args:
+            name (str): The model name.
+            model_data_prefix (str): The S3 prefix where all the models artifacts (.tar.gz)
+                in a Multi-Model endpoint are located
+            model (sagemaker.Model): The Model object that would define the
+                SageMaker model attributes like vpc_config, predictors, etc.
+                If this is present, the attributes from this model are used when
+                deploying the ``MultiDataModel`` and local attributes are ignored.
+            image (str): A Docker image URI. If not provided, `model.image` is used instead.
+                (default: None)
+            role (str): An AWS IAM role (either name or full ARN). The Amazon
+                SageMaker training jobs and APIs that create Amazon SageMaker
+                endpoints use this role to access training data and model
+                artifacts. After the endpoint is created, the inference code
+                might use the IAM role if it needs to access some AWS resources.
+                It can be null if this is being used to create a Model to pass
+                to a ``PipelineModel`` which has its own Role field. If not provided,
+                `model.image` is used instead. (default: None)
+            sagemaker_session (sagemaker.session.Session): A SageMaker Session
+                object, used for SageMaker interactions (default: None). If not
+                specified, one is created using the default AWS configuration
+                chain.
+            **kwargs: Keyword arguments passed to the ``Model`` initializer.
+        """
+        # Validate path
+        if not model_data_prefix.startswith("s3://"):
+            raise ValueError(
+                'Expecting S3 model prefix beginning with "s3://". Received: "{}"'.format(
+                    model_data_prefix
+                )
+            )
+        self.name = name
+        self.model_data_prefix = model_data_prefix
+        self.model = model
+        self.container_mode = MULTI_MODEL_CONTAINER_MODE
+        self.sagemaker_session = sagemaker_session or Session()
+        self.s3_client = self.sagemaker_session.boto_session.client("s3")
+
+        # Set the ``Model`` parameters if the model parameter is not specified
+        if not self.model:
```

Comment:
I think it would be better to just raise a ValueError if model and other parameters like image, role, etc. are provided, instead of silently ignoring them.

Reply:
Sure will add the validation